packages/perseus/src/widgets/interactive-graphs/axis-ticks.tsx

```diff
@@ -11,87 +13,78 @@ const tickStyle: React.CSSProperties = {
+    // even if they are outside the graph's range.
+    if (graphInfo.range[0][0] > 0) {
+        // If the graph is on the positive side of the x-axis, lock the ticks to the left side of the graph
+        xPointOnAxis = graphInfo.range[0][0];
+    }
+    if (graphInfo.range[0][1] < 0) {
+        // If the graph is on the negative side of the x-axis, lock the ticks to the right side of the graph
+        xPointOnAxis = graphInfo.range[0][1];
+    }
 
-const YGridTick = ({
-    y,
-    gridStep,
-    tickStep,
-}: {
-    y: number;
-    gridStep: number;
-    tickStep: number;
-}) => {
-    const pointOnAxis: vec.Vector2 = [0, y];
+    const pointOnAxis: vec.Vector2 = [xPointOnAxis, y];
     const [[xPosition, yPosition]] = useTransformVectorsToPixels(pointOnAxis);
 
+    // If the tick is on the edge of the graph's range, don't render it
+    if (
+        yPosition === -graphInfo.height ||
+        yPosition === graphInfo.height + 20
+    ) {
+        return null;
+    }
+
+    // Position of the start of the tick
+    const x1 = xPosition - tickSize / 2;
+    const y1 = yPosition;
+
+    // Position of the end of the tick
+    const x2 = xPosition + tickSize / 2;
+    const y2 = yPosition;
+
     return (
         <g className="y-axis-ticks">
-            <line
-                x1={xPosition - tickSize / 2}
-                y1={yPosition}
-                x2={xPosition + tickSize / 2}
-                y2={yPosition}
-                style={tickStyle}
-            />
-            {showTickLabel(gridStep, tickStep, y) && (
-                <text
-                    height={20}
-                    width={50}
-                    textAnchor="end"
-                    x={xPosition - 10}
-                    y={yPosition + 5}
-                >
-                    {y.toString()}
-                </text>
-            )}
+            <line x1={x1} y1={y1} x2={x2} y2={y2} style={tickStyle} />
         </g>
     );
 };
 
-const XGridTick = ({
-    x,
-    gridStep,
-    tickStep,
-}: {
-    x: number;
-    gridStep: number;
-    tickStep: number;
-}) => {
-    const pointOnAxis: vec.Vector2 = [x, 0];
+const XGridTick = ({x, graphInfo}: {x: number; graphInfo: GraphDimensions}) => {
+    let yPointOnAxis = 0;
+    // If the graph is zoomed in, we want to make sure the ticks are still visible
+    // even if they are outside the graph's range.
+    if (graphInfo.range[1][0] > 0) {
+        // If the graph is on the positive side of the x-axis, lock the ticks to the left side of the graph
+        yPointOnAxis = graphInfo.range[1][0];
+    }
+    if (graphInfo.range[1][1] < 0) {
+        // If the graph is on the negative side of the x-axis, lock the ticks to the right side of the graph
+        yPointOnAxis = graphInfo.range[1][1];
+    }
+
+    const pointOnAxis: vec.Vector2 = [x, yPointOnAxis];
     const [[xPosition, yPosition]] = useTransformVectorsToPixels(pointOnAxis);
 
+    // If the tick is on the edge of the graph's range, don't render it
```

Comment:
This one is probably worth a new test, but it's an extreme edge case. I'll get a test made up for it.